### DISCUSSION TITLE
[CLI] Attach cloudfront to S3 bucket

### DISCUSSION THREAD

**Discussion by VaZark** [aws/copilot-cli#5150] [Ideas]
When using django on AWS, it is standard practice to serve static files from S3 behind cloudfront. (user-uploaded media files are stored with private ACL)

It would great if cloudfront distribution can be provisioned for the S3 addon right from the CLI and/or manifest.

**Comment by KollaAdithya** [comment]
Hello @VaZark !

Doesn't `copilot static site`[(doc)](https://aws.github.io/copilot-cli/docs/manifest/static-site/) serve your usecase. Copilot static site creates an S3 bucket and uploading your chosen local files to that bucket, and generating a CloudFront distribution with the S3 bucket as the origin.

Is this the requirement with S3 addon created by copilot, cloudfront distribution should be provisioned from the S3 bucket addon ?

can you provide context on your use case ?

**Reply by VaZark** [reply]
Gladly. It's as you put it - cloudfront for the s3 addon.

When serving server-side rendered applications, additional files like script files, external stylesheets can be cached and served to all users. For user specific media files like pdf or images, we provide a signed link.

So now at the server can just provide signed and unsigned links to documents hosted on S3.

This pretty std practice for django apps.

A popular django lib that provides support for cloud storage : https://django-storages.readthedocs.io/en/latest/backends/amazon-S3.html 

Blogpost on how these are usually setup: https://testdriven.io/blog/storing-django-static-and-media-files-on-amazon-s3/

**Comment by KollaAdithya** [comment]
Hey @VaZark !
I just created a new issue #5152. upvote this issue to help us priortize
